R/utils.R

```diff
@@ -0,0 +1,65 @@
+#' Code borrowed from SLOPE package for generating datasets
+#' @export
+randomProblem <-
+  function(n = 1000,
+           p = 100,
+           q = 0.2,
+           n_groups = NULL,
+           n_targets = if (match.arg(response) == "multinomial") 3 else 1,
+           density = 1,
+           amplitude = if (match.arg(response) == "poisson") 1 else 3,
+           alpha = 1,
+           response = c("gaussian", "binomial", "poisson", "multinomial"),
+           rho = 0) {
+  m <- n_targets
+
+  if (density == 1) {
+    x <- matrix(stats::rnorm(n*p), n)
+  } else {
+    x <- Matrix::rsparsematrix(n, p, density)
+  }
+
+  if (rho > 0)
+    x <- x + sqrt(rho/(1 - rho)) * matrix(stats::rnorm(n), n, p)
```

Comment:
```suggestion
    x <- sqrt(1-rho) * x + sqrt(rho) * matrix(stats::rnorm(n), n, p)
```
This will make the varians of the elments one and correlation rho.